src/python/tests/test_configuration_manager.py

```diff
@@ -0,0 +1,710 @@
+                                          'type': 'float',
+                                          'default': '1.1'}})
+        assert ('ValueError: Invalid entry_val for entry_name "type" for item_name info. valid: ' +
+                "['boolean', 'integer', 'string', 'IPv4', " +
+                "'IPv6', 'X509 certificate', 'password', 'JSON']") in str(error_exec)
+        await self.teardown_method()
+
+        await self.setup_method()
+        with pytest.raises(ValueError) as error_exec:
+            await create_category(category_name='INTEGER', category_description='INTEGER type',
+                                  category_value={
+                                      'info': {
+                                          'description': 'INTEGER type with default 1',
+                                          'type': 'INTEGER',
+                                          'default': '1'}})
+        assert ('ValueError: Invalid entry_val for entry_name "type" for item_name info. valid: ' +
+                "['boolean', 'integer', 'string', 'IPv4', " +
+                "'IPv6', 'X509 certificate', 'password', 'JSON']") in str(error_exec)
+        await self.teardown_method()
+
+    async def test_create_category_invalid_entry_value(self):
+        """
+        Test cases where value is set to the actual "value" rather than the string of the value
+        :Assert:
+            1. Assert TypeError when type is set to bool rather than 'boolean'
+            2. Assert TypeError when default is set to False rather than 'False'
+        """
+        await self.setup_method()
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean', category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': bool,
+                                      'default': 'False'
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name " +
+                "info and entry_name type") in str(error_exec)
+        await self.teardown_method()
+
+        await self.setup_method()
+        with pytest.raises(TypeError) as error_exec:
+            await create_category(category_name='boolean',
+                                  category_description='boolean type',
+                                  category_value={'info': {
+                                      'description': 'boolean type with default False',
+                                      'type': 'boolean',
+                                      'default': False
+                                  }})
+        assert ("TypeError: entry_val must be a string for item_name " +
+                "info and entry_name default") in str(error_exec)
+        await self.teardown_method()
```

Comment:
break into 2 tests `test_create_category_invalid_entry_value_type` and `test_create_category_invalid_entry_value`